modules/apps/frontend-css/frontend-css-variables-web/src/main/java/com/liferay/frontend/css/variables/web/internal/theme/ThemeCSSVariableDescriptionsServiceTrackerCustomizer.java

```diff
@@ -0,0 +1,192 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.frontend.css.variables.web.internal.theme;
+
+import com.liferay.frontend.css.variables.CSSVariableDescription;
+import com.liferay.frontend.css.variables.CSSVariableType;
+import com.liferay.frontend.css.variables.web.internal.CSSVariableDescriptionImpl;
+import com.liferay.petra.string.StringPool;
+import com.liferay.portal.kernel.json.JSONException;
+import com.liferay.portal.kernel.json.JSONFactory;
+import com.liferay.portal.kernel.json.JSONObject;
+import com.liferay.portal.kernel.log.Log;
+import com.liferay.portal.kernel.log.LogFactoryUtil;
+import com.liferay.portal.kernel.util.StringBundler;
+import com.liferay.portal.kernel.util.StringUtil;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import javax.servlet.ServletContext;
+
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.ServiceReference;
+import org.osgi.util.tracker.ServiceTrackerCustomizer;
+
+/**
+ * @author Iván Zaera Avellón
+ */
+public class ThemeCSSVariableDescriptionsServiceTrackerCustomizer
+	implements ServiceTrackerCustomizer
+		<ServletContext, Map<String, CSSVariableDescription>> {
+
+	public ThemeCSSVariableDescriptionsServiceTrackerCustomizer(
+		BundleContext bundleContext, JSONFactory jsonFactory) {
+
+		_bundleContext = bundleContext;
+		_jsonFactory = jsonFactory;
+	}
+
+	@Override
+	public Map<String, CSSVariableDescription> addingService(
+		ServiceReference<ServletContext> serviceReference) {
+
+		Map<String, CSSVariableDescription> cssVariablesDescriptions =
+			new ConcurrentHashMap<>();
+
+		modifiedService(serviceReference, cssVariablesDescriptions);
+
+		return cssVariablesDescriptions;
+	}
+
+	@Override
+	public void modifiedService(
+		ServiceReference<ServletContext> serviceReference,
+		Map<String, CSSVariableDescription> cssVariablesDescriptions) {
+
+		ServletContext servletContext = _bundleContext.getService(
+			serviceReference);
+
+		try {
+			InputStream is = servletContext.getResourceAsStream(
+				"/WEB-INF/css-variables.json");
```

Comment:
Let's make this configurable and give it a nice default, just in case...

Reply:
Let's call the default `tokens.json` or something like that so we don't tie this to a particular implementation.

Reply:
You mean the file name? Why would we want to make it configurable? :thinking:

Reply:
Because we often shoot ourselves in the foot and change the name later or do some other weird stuff.

Just define a `tokensPath` or something like that and make `tokens.json` the default in case nothing is passed. We can then generate dev tooling for this without hardcoding "css-variables.json"  everywhere in different projects and forcing users down that name.

Reply:
I agree with the renaming part :sweat_smile: . 

The problem with making it configurable is that once a tool/portal version starts using a certain location making configuration changes can only lead to disaster because then you have to care about tooling using old names, new names, and how to setup the portal. 

Also, you may need to deploy WAR files using the old location for the file and WARs using the new one, but you may only configure one value in the portal, unless we add support for multiple places.

All other configuration files (MANIFEST.MF, OSGi metatype, package.json and so on) are placed in well-known locations for that reason.

I'll extract it to a constant/property, anyway, though I have used constants in the past for the very same reasons you are stating and they were removed in favour of hard coded strings in revisions...

Reply:
By configurable I mean in the `package.json`, not as a Portal Setting

Reply:
Just to confirm: you mean adding a known key in the `liferayTheme` section of the `package.json` to specify where this file is stored inside the WAR?